n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,163 @@
+from eko import run_dglap
+
+
+def evolve_fit(conf_folder):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit 
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+    """
+    usr_path = pathlib.Path(conf_folder)
+    initial_PDFs_dict = load_fit(usr_path)
+    eko, theory, op = construct_eko_for_fit(usr_path)
+    info = gen_info.create_info_file(theory, op, 1, info_update={})  # to be changed
+    dump_info_file(usr_path, info)
+    utils.fix_info_path(usr_path)
+    for replica in initial_PDFs_dict.keys():
+        evolved_block = evolve_exportgrid(initial_PDFs_dict[replica], eko, theory, op)
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+        # fixing_replica_path
+        utils.fix_replica_path(usr_path, int(replica.removeprefix("replica_")))
+    # remove folder
+    os.rmdir(usr_path / "nnfit" / usr_path.stem)
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+    
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for subdir, dirs, files in os.walk(nnfitpath):
+        for dir in dirs:
+            replica_list.append(dir)
+    replica_list.remove("input")
+    # remove the eventual evolution folder
+    try:
+        replica_list.remove(usr_path.stem)
+    except:
+        pass
+    pdf_dict = {}
+    for replica in replica_list:
+        rep_path = pathlib.Path(replica) / (usr_path.stem + ".exportgrid")
+        yaml_file = nnfitpath / rep_path.relative_to(rep_path.anchor)
+        with yaml_file.open() as fp:
+            data = yaml.safe_load(fp)
+        pdf_dict[replica] = data
+    return pdf_dict
+
+
+# Temporary solution. Then it will be loaded from the theory itself
+def construct_eko_for_fit(usr_path):
+    """
+    Construct the eko operator needed for evolution of fitted pdfs
+
+    Parameters
+    ----------
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+    Returns
+    -------
+        : eko.output.Output
+        eko operator
+        : dict
+        theory runcard
+        : dict
+        operator runcard
+    """
+    # read the runcard
+    my_runcard = utils.read_runcard(usr_path)
+    # theory_card construction
+    theory = Loader().check_theoryID(my_runcard["theory"]["theoryid"]).get_description()
+    theory.pop("FNS")
+    t_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    # construct operator card
+    op_x_grid = utils.generate_x_grid()
+    q2_grid = utils.generate_q2grid(theory["Q0"], 1.0e5)
+    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": op_x_grid})
```

Comment:
I would give the possibility to change other options of the operator card at least the `Q2_grid`, `x_grid`, `n_integration_cores` and `ev_op_iterations`